backend/test_execution_engine.py

```
@@ -1085,6 +1174,355 @@ def _calculate_overall_status(self, passed: int, failed: int, total: int) -> str
+            conn = await asyncpg.connect(self.database_url)
+            try:
+                # Get generator method from database configuration
+                result = await conn.fetchrow(query, suite_name)
+
+                if result and result[COLUMN_GENERATOR_METHOD]:
+                    generator_method_name = result[COLUMN_GENERATOR_METHOD]
+
+                    if not self._validate_generator_method_security(generator, generator_method_name):
+                        logger.error(f"Security validation failed for generator method: {generator_method_name}")
+                        logger.warning(f"Falling back to safe default method for suite '{suite_name}'")
+                        return await getattr(generator, "generate_integration_tests")()
```

Ruff (0.12.2)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)